### PR TITLE
Labels vcher dscptn

### PR DESCRIPTION
**Pull request by francisguchie**
Description of the issue/feature this PR addresses:
Clients want to see two fields voucher (required) and description (null-able) at all loan repayments and savings deposits and withdrawals
Current behavior before PR:
at the moment these two fields are not available 
Desired behavior after PR is merged:
these two fields are displayed as desired

--

### CODE REVIEW

File: fineract-provider/src/main/resources/sql/migrations/core_db/V331__modify_payment_detail_add_vcherdescription_columns.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+--
+-- Unless required by applicable law or agreed to in writing,
+-- software distributed under the License is distributed on an
+-- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+-- KIND, either express or implied. See the License for the
+-- specific language governing permissions and limitations
+-- under the License.
+--
+ 
+ALTER TABLE `m_payment_detail` ADD COLUMN `voucher_details` VARCHAR(50) NULL DEFAULT NULL;
```
Comment (ottoman91) [inline fineract-provider/src/main/resources/sql/migrations/core_db/V331__modify_payment_detail_add_vcherdescription_columns.sql]: @francisguchie can this be renamed as voucher_number ?

**Comment by francisguchie** [comment]
@ottoman91  this migration sql  V327__creditbureau_configuration.sql should be as follows to avoid  a migration crush ( Either comment out the table creations or remove the creations)


/* -- credit bureau tables 
CREATE TABLE `m_creditbureau` (
  `id` bigint(20) NOT NULL AUTO_INCREMENT,
  `name` varchar(100) NOT NULL,
  `product` varchar(100) NOT NULL,
  `country` varchar(100) NOT NULL,
  `implementationKey` varchar(100) NOT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `unique impl` (`name`,`product`,`country`,`implementationKey`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
*/

/*CREATE TABLE `m_organisation_creditbureau` (
  `id` bigint(20) NOT NULL AUTO_INCREMENT,
  `alias` varchar(50) NOT NULL,
  `creditbureau_id` bigint(20) NOT NULL,
  `is_active` tinyint(4) NOT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `morgcb` (`alias`,`creditbureau_id`),
  KEY `orgcb_cbfk` (`creditbureau_id`),
  CONSTRAINT `orgcb_cbfk` FOREIGN KEY (`creditbureau_id`) REFERENCES `m_creditbureau` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
*/


/*
CREATE TABLE `m_creditbureau_configuration` (
  `id` bigint(20) NOT NULL AUTO_INCREMENT,
  `configkey` varchar(50) DEFAULT NULL,
  `value` varchar(50) DEFAULT NULL,
  `organisation_creditbureau_id` bigint(20) DEFAULT NULL,
  `description` varchar(50) DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `mcbconfig` (`configkey`,`organisation_creditbureau_id`),
  KEY `cbConfigfk1` (`organisation_creditbureau_id`),
  CONSTRAINT `cbConfigfk1` FOREIGN KEY (`organisation_creditbureau_id`) REFERENCES `m_organisation_creditbureau` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
*/

/*
CREATE TABLE `m_creditbureau_loanproduct_mapping` (
  `id` bigint(20) NOT NULL AUTO_INCREMENT,
  `organisation_creditbureau_id` bigint(20) NOT NULL,
  `loan_product_id` bigint(20) NOT NULL,
  `is_creditcheck_mandatory` tinyint(1) DEFAULT NULL,
  `skip_creditcheck_in_failure` tinyint(1) DEFAULT NULL,
  `stale_period` int(11) DEFAULT NULL,
  `is_active` tinyint(1) DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `cblpunique_key` (`organisation_creditbureau_id`,`loan_product_id`),
  KEY `fk_cb_lp2` (`loan_product_id`),
  CONSTRAINT `cblpfk2` FOREIGN KEY (`organisation_creditbureau_id`) REFERENCES `m_organisation_creditbureau` (`id`),
  CONSTRAINT `fk_cb_lp2` FOREIGN KEY (`loan_product_id`) REFERENCES `m_product_loan` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

*/

--  modify `m_portfolio_command_source` command

ALTER TABLE `m_portfolio_command_source`
	ADD COLUMN `creditbureau_id` BIGINT(20) NULL DEFAULT NULL AFTER `transaction_id`,
	ADD COLUMN `organisation_creditbureau_id` BIGINT(20) NULL DEFAULT NULL AFTER `creditbureau_id`;


-- permissions
INSERT INTO `m_permission` (`grouping`, `code`, `entity_name`, `action_name`, `can_maker_checker`) VALUES ('configuration', 'CREATE_CREDITBUREAU_LOANPRODUCT_MAPPING', 'CREDITBUREAU_LOANPRODUCT_MAPPING', 'CREATE', 0);
INSERT INTO `m_permission` (`grouping`, `code`, `entity_name`, `action_name`, `can_maker_checker`) VALUES ('configuration', 'CREATE_ORGANISATIONCREDITBUREAU', 'ORGANISATIONCREDITBUREAU', 'CREATE', 0);
INSERT INTO `m_permission` (`grouping`, `code`, `entity_name`, `action_name`, `can_maker_checker`) VALUES ('configuration', 'UPDATE_ORGANISATIONCREDITBUREAU', 'ORGANISATIONCREDITBUREAU', 'UPDATE', 0);
INSERT INTO `m_permission` (`grouping`, `code`, `entity_name`, `action_name`, `can_maker_checker`) VALUES ('configuration', 'UPDATE_CREDITBUREAU_LOANPRODUCT_MAPPING', 'CREDITBUREAU_LOANPRODUCT_MAPPING', 'UPDATE', 0);

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/SavingsApiConstants.java [hunk 2]
```diff
@@ -124,6 +124,8 @@
     public static final String routingCodeParamName = "routingCode";
     public static final String receiptNumberParamName = "receiptNumber";
     public static final String bankNumberParamName = "bankNumber";
+    public static final String voucherParamName = "voucher";
```
Comment (ottoman91) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/SavingsApiConstants.java]: @francisguchie to follow the proper naming conventions for the variables, i would suggest the following: voucherNumber . Note the camelCase font that has been used in all variables declared.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/SavingsApiConstants.java [hunk 3]
```diff
@@ -124,6 +124,8 @@
...
+    public static final String paymentdetaildescriptionParamName = "paymentdetaildescription"; // for payment description
```
Comment (ottoman91) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/SavingsApiConstants.java]: @francisguchie use paymentDetailDescription as the variable name, again following the camelCase convention

**Pull request closed without merging**